src/inventory/InventoryFactory.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\inventory;
+
+use Closure;
+use pocketmine\block\inventory\AnvilInventory;
+use pocketmine\block\inventory\BrewingStandInventory;
+use pocketmine\block\inventory\EnchantInventory;
+use pocketmine\block\inventory\FurnaceInventory;
+use pocketmine\block\inventory\HopperInventory;
+use pocketmine\block\inventory\LoomInventory;
+use pocketmine\crafting\FurnaceType;
+use pocketmine\network\mcpe\protocol\types\inventory\UIInventorySlotOffset;
+use pocketmine\network\mcpe\protocol\types\inventory\WindowTypes;
+use pocketmine\utils\AssumptionFailedError;
+use pocketmine\utils\SingletonTrait;
+use pocketmine\utils\Utils;
+use function array_key_exists;
+
+final class InventoryFactory{
```

Comment:
This doesn't belong in the `inventory` namespace. `network\mcpe` or `network\mcpe\convert` would be more appropriate.